Solaris has yet another type of fd_set::fds_bits. This should be offered to upstream.

--- net-snmp-5.9.4/configure.d/config_project_types
+++ net-snmp-5.9.4/configure.d/config_project_types
@@ -66,7 +66,7 @@ netsnmp_save_CFLAGS=$CFLAGS
 CFLAGS="$CFLAGS -Werror"
 
 AC_MSG_CHECKING([for the type of fd_set::fds_bits])
-for type in __fd_mask __int32_t unknown; do
+for type in __fd_mask __int32_t "unsigned long"; do
   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
 #include <sys/select.h>
 #include <stddef.h>
